src/jinja2/environment.py

```diff
@@ -898,7 +902,7 @@ def from_string(self, source, globals=None, template_class=None):
     def make_globals(self, d):
         """Return a dict for the globals."""
         if not d:
-            return self.globals
+            return self.globals.copy()
```

Comment:
I remember from a previous PR that we decided this was not necessary since it took up extra space. But when fixing this bug, I ran into unintended side effects as a result.